### ISSUE TITLE
New stable release

### ISSUE THREAD

**Issue by FabioPedretti** [xiph/theora#17]
Some projects are removing Theora support for some reasons, including the fact the last official release was too many years ago, and also missing some important fixes (ome references below). Linux distros also needs to include many patches to build on recent environments: https://salsa.debian.org/multimedia-team/libtheora/-/tree/master/debian/patches

Is there a plan to have a new release with the accumulated fixes? A new release with the [Ptalarbvorm](https://people.xiph.org/~xiphmont/demo/theora/demo9.html) would be even better. 💯

---

DaemonEngine: https://github.com/DaemonEngine/Daemon/pull/391
> I noticed the Theora project seems kind of dead. When compiling, I saw a gcc warning that the decoder code indexes out of bounds. Although this was [fixed in 2012](https://github.com/xiph/theora/commit/0ae66d565e6bead8604d312bc1a4e9dccf245c88), there hasn't been a new release since 2010 so we didn't have the fix.

Google Chrome: https://chromestatus.com/feature/5158654475239424
> The Theora video codec was first released in 2004 and had its last official release in 2011.

Mozilla Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1860492

**Comment by jpsaikali** [comment]
VP8 only supports 4:2:0, while Theora also supports 4:2:2 and 4:4:4. Although both VP9 and AV1 support all three ratios and HDR, there is [legal uncertainty](https://www.sisvel.com/licensing-programmes/audio-and-video-coding-decoding/) surrounding their royalty-free statuses. Therefore, I personally think that there should be security updates for Theora at least until H.264 becomes a free format in 5 years at most (I think). By then, I think higher resolution displays would become more ubiquitous, in which the improvements of 4:2:2 and 4:4:4 over 4:2:0 would become less noticeable.

**Comment by Neustradamus** [comment]
Important to have new release...

**Comment by FOSSPCM** [comment]
I was hoping there would be a new, stable 1.2 release. It looks like Xiph is prioritizing Opus, FLAC, and rav1e right now; and they were working on Dalaa for a while before AV1 became a thing.

From what I've read, the reason why some projects are dropping Theora is because of lack of development, which caused worry in the wake of the WebP vulnerability (there is no security vulnerability in Theora that we know of, but some projects don't want to take chances on something that hasn't been updated in over ten years), and because there is little usage of Theora today. That's a shame too, because Theora is historically significant, is the only reliably free video format without looming patent claims, and as @jpsaikali mentioned, it has features not found in VP8. I hate to see Theora appear so unloved.

Is there anyone at Xiph who can make an update to Theora happen?

**Comment by FabioPedretti** [comment]
You may as well have a look at [MPEG-4 Visual](https://en.wikipedia.org/wiki/MPEG-4_Part_2) / [Xvid](https://en.wikipedia.org/wiki/Xvid), [All worldwide patents have expired except for 2 patents still active in Brazil](https://meta.wikimedia.org/wiki/Have_the_patents_for_MPEG-4_Visual_expired_yet%3F).

**Comment by tdaede** [comment]
I can ship a new release if desired. The current 1.2 encoder does have some quality regressions but also improvements, but likely the compilation fixes are more important than those.

If someone would like to become (co)-maintainer, and preferably has some history in open source projects, let us know in #theora or #xiph on libera.chat!

**Comment by FOSSPCM** [comment]
@tdaede I agree. For right now, getting the compilation fixes in there would be nice, just so those continuing to use Theora (Pale Moon, VLC...) still have something, and we can say Theora isn't "dead."

As for version 1.2, I found an old to-do list of things desired for it:
- (done) Activity masking (greatly improved perceptual quality)
- (done) Rename speedlevel 2 to 3 and provide a new speedlevel 2
- (done) ARM4/6/7/Neon assembly support
- (done) C64x+ support
- Fix SAD-for-SATD in intra analysis from Ralph's speed level patch.
- Testing testing testing
- Decoder speedups (improvements to ARM/c64x platform support)
- Encoder speedups (additional short-cutting in fast modes)
- Temporal RDO
- Quantizer matrix improvements (Shift the range up to make higher rates achievable)
- Mode selection improvements (split 3qi roles?)
- Codebook re-optimization (Greg has code for this; waiting for other changes to settle)
- Review open tickets
- Review patches from the major distros

I'm not sure know who "Greg" and "Ralph" are, but we can worry about this at a later date.

I don't feel comfortable being co-maintainer at this time (not as much history contributing to open source projects, although I've been a big advocate for them since 2014 or so; and I haven't had my GitHub account long). I'd rather build up a reputation first. Can't take too many chances since [the XZ incident](https://en.wikipedia.org/wiki/XZ_Utils_backdoor),

Having said that, I just made a fork, and I can contribute with pull requests where I can!

Also, sorry about the late reply.

**Comment by Neustradamus** [comment]
Never too late!

**Comment by petterreinholdtsen** [comment]
A request for a new version is also on https://gitlab.xiph.org/xiph/theora/-/issues/2319 .  I am working on wrapping up a new release .

**Comment by Neustradamus** [comment]
@petterreinholdtsen: Good news, thanks in advance!
And like I have said before, it is never too late to do it because a new version is very important :)

**Comment by petterreinholdtsen** [comment]
Please test the code in the master branch on https://gitlab.xiph.org/xiph/theora/ and report problem there if you find any.  I plan to wrap up the new release the coming weekend.

**Comment by Neustradamus** [comment]
@ePirat: Good job, thanks for 1.2.0 (2025-03-29):
- https://github.com/xiph/theora/releases/tag/v1.2.0
